templates/default/deploy2glassfish_hook.sh.erb

```diff
@@ -1,89 +1,157 @@
+    keytool -importkeystore -destkeystore keystore.jks -srckeystore pkcs.p12 -srcstoretype PKCS12 -alias glassfish-instance -srcstorepass $KEYSTOREPW -deststorepass $KEYSTOREPW -destkeypass $KEYSTOREPW
+    openssl pkcs12 -export -in ${working_dir}/fullchain.pem -inkey ${working_dir}/privkey.pem -out pkcs.p12 -name s1as -password pass:$KEYSTOREPW
+    keytool -delete -keystore keystore.jks -alias s1as -srcstorepass $KEYSTOREPW -deststorepass $KEYSTOREPW -destkeypass $KEYSTOREPW
+    keytool -importkeystore -destkeystore keystore.jks -srckeystore pkcs.p12 -srcstoretype PKCS12 -alias s1as -srcstorepass $KEYSTOREPW -deststorepass $KEYSTOREPW -destkeypass $KEYSTOREPW
+
+    keytool -export -alias glassfish-instance -file glassfish-instance.cert -keystore keystore.jks -storepass $KEYSTOREPW
+    keytool -export -alias s1as -file s1as.cert -keystore keystore.jks -storepass $KEYSTOREPW
+
+    # Update Glassfish CA certs
+    keytool -delete -keystore cacerts.jks -alias s1as -storepass $KEYSTOREPW
+    keytool -import -noprompt -alias s1as -file s1as.cert -keystore cacerts.jks -storepass $KEYSTOREPW
+    keytool -delete -keystore cacerts.jks -alias glassfish-instance -storepass $KEYSTOREPW
+    keytool -import -noprompt -alias glassfish-instance -file glassfish-instance.cert -keystore cacerts.jks -storepass $KEYSTOREPW
+
+    # Convert cacerts to pem format
+    keytool -importkeystore -srckeystore cacerts.jks -destkeystore cacerts.p12 -srcstoretype jks -deststoretype pkcs12 -noprompt -srcstorepass $KEYSTOREPW -deststorepass $KEYSTOREPW
+    openssl pkcs12 -in cacerts.p12 -out cacerts.pem -passin pass:$KEYSTOREPW
+
+    cp -f cacerts.jks ${SPARK_CONF_DIR}/cacerts.jks
+    mv -f cacerts.pem ${SPARK_CONF_DIR}/cacerts.pem
+    mv -f keystore.jks ${GF_DOMAIN}/config/keystore.jks
+    mv -f cacerts.jks ${GF_DOMAIN}/config/cacerts.jks
+
+    rm -f cacerts.p12
+
+    pushd ${GF_DOMAIN}/config
+    chown ${GF_USER}:${GF_GROUP} keystore.jks
+    chmod 600 keystore.jks
+
+    chown ${GF_USER}:${GF_GROUP} cacerts.jks                                                                                                                                                                                          [12/475]
+    chmod 655 cacerts.jks
+
+    systemctl restart glassfish-$GF_DOMAIN_NAME
+    echo "Restarting glassfish..."
+    yes | ${GF_DOMAIN}/bin/domain1_asadmin disable-secure-admin
+    yes | ${GF_DOMAIN}/bin/domain1_asadmin enable-secure-admin
+}
+
+_deploy_aws_certificate_manager(){
+    _log_info "Deploying AWS certificate manager"
+    if [ -z "$certificate_arn" ]; then
+        _log_error "You have to supply AWS Certificate Manager certificate ARN"
+        exit 1
+    fi
+    aws_region=$(echo $certificate_arn | awk -F ":" '{ print $4}')
+    _log_info "AWS: Importing to $certificate_arn - Region: $aws_region"
+    aws --region ${aws_region} acm import-certificate \
```

Comment:
It this the only permissions we need to add to the instance profile then?

Reply:
I have them documented in the JIRA https://hopsworks.atlassian.net/browse/CLOUD-616?focusedCommentId=16139

Reply:
The process is not very straightforward but I explain why on my first comment

Reply:
aha, ok, but why do we need to provide the certificate_arn to start with it? 
```
The [Amazon Resource Name (ARN)](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) of an imported certificate to replace. To import a new certificate, omit this field.
```
https://docs.aws.amazon.com/cli/latest/reference/acm/import-certificate.html